internal/k8s/helpers.go

```diff
@@ -83,3 +89,81 @@ func isStatusEqual(objA, objB interface{}) bool {
 	}
 	return false
 }
+
+// IsObjectEqual checks if objects received during update are equal.
+//
+// Make an attempt to avoid comparing full objects since it can be very CPU intensive.
+// Prefer comparing Generation when only interested in spec changes.
+func IsObjectEqual(old, new interface{}) (bool, error) {
+
+	// Fast path for any object: when ResourceVersions are equal, the objects are equal.
+	// NOTE: This optimizes the case when controller-runtime executes full sync and sends updates for all objects.
+	if isResourceVersionEqual(old, new) {
+		return true, nil
+	}
+
+	switch old := old.(type) {
```

Comment:
need to add handling for `Namespace` and `GRPCRoute`. In the case of Namespace, the Gateway API processor actually looks at the labels, so those need to taken into account for the comparison. GRPCRoute _should_ be like the other two Gateway API route types.

Reply:
Thank you for spotting this, I've now added these types!